### ISSUE TITLE
Question on requires_grad

### ISSUE THREAD

**Issue by yangshao** [GriffinLiang/vrd-dsr#8]
Hi, I'm trying to run this code on my own data, but i don't understand why do you set the object classification layer's requires_grad as False?
 network.set_trainable(self.fc_obj, requires_grad=False)

**Comment by GriffinLiang** [comment]
Because I use proposal.pkl to acquire the object classification confidence. You can change it if you need.

**Comment by yangshao** [comment]
@GriffinLiang So this layer(self.fc_obj) is only used in the relation testing instead of training?

**Comment by yangshao** [comment]
@GriffinLiang when I try to also train the objects label at the same time, the training loss doesn't decrease anymore. The only change i do is set fc_obj's requires_grad = True and set the loss as object classification cross entropy loss.  Have you ever tried training object labels? Thanks.

**Comment by yangshao** [comment]
I figured out the reason: i need to add new layers to opt_params...
